qiskit/quantum_info/operators/dihedral/dihedral.py

```diff
@@ -0,0 +1,495 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+CNOTDihedral operator class.
+
+Methods for working with the CNOT-dihedral group.
+
+Example:
+
+  from dihedral import CNOTDihedral
+  g = CNOTDihedral(3)  # create identity element on 3 qubits
```

Comment:
This shouldn't be API for how you use these operators since it is very different to every other operator. This kind of construction can either be done as a circuit, or using compose.

Eg:
```python
circ = QuantumCircuit(3)
circ.cx(0, 1)
circ.x(2)
circ.t(1)
circ.t(1)
circ.t(1)  # dont think gates support power atm
g = CNOTDihedral(circ)
```

or 
```
g = CNOTDihedral(num_qubits=3)
g = g.compose(CXGate(), [0, 1])
g = g.compose(XGate(), [2])
g = g.compose(CNOTDIhedral(TGate()) ** 3, [1])
```
Compose could be replaced with binary op, but we are going to deprecate use of `@` for `compose` soon so it will be a different op.

Reply:
I cannot give the compose example now, since compose method does not support qargs yet.